catalog/plugin/catalog-plugin-defaultsecurityattributevalues/src/main/java/org/codice/alliance/catalog/plugin/defaultsecurity/DefaultSecurityAttributeValuesPlugin.java

```diff
@@ -205,57 +230,140 @@ public DeleteRequest process(DeleteRequest input)
     return input;
   }
 
-  private void addMapping(String userAttribute, String metacardAttribute) {
-    Set<String> metacardAttributes =
-        SYS_HIGH_TO_METACARD_ATTRIBUTE_MAPPING.get(userAttribute) != null
-            ? SYS_HIGH_TO_METACARD_ATTRIBUTE_MAPPING.get(userAttribute)
-            : new LinkedHashSet<>();
-    metacardAttributes.add(metacardAttribute);
-    SYS_HIGH_TO_METACARD_ATTRIBUTE_MAPPING.put(userAttribute, metacardAttributes);
-  }
-
   /**
-   * Retrieves the system high attributes and the mapping of the system attribute names to metacard
-   * security markings. Returns a hash map of the metacard markings to the value of its
-   * corresponding system attribute.
+   * Retrieves the system high attributes for the system high attribute names in the properties
+   * parameter.
    *
-   * @return Map of the metacard security markings to the value of their corresponding system high
-   *     attribute.
+   * @param properties a map of configuration keys to system high attribute names
+   * @return a map of metacard attribute names to corresponding system attribute values, if each
+   *     system attribute is present
+   * @throws DefaultSecurityAttributeValuesPluginException when there is an unexpected error
+   *     retrieving the system attribute values
    */
-  private Map<String, Attribute> getHighwaterSecurityMarkings() {
-    Map<String, Attribute> securityMarkings = new HashMap<>();
-    Subject system =
-        org.codice.ddf.security.common.Security.getInstance().runAsAdmin(subjectSupplier::get);
-    SecurityAssertion assertion = system.getPrincipals().oneByType(SecurityAssertion.class);
-    List<AttributeStatement> attributeStatements = assertion.getAttributeStatements();
-    for (AttributeStatement curStatement : attributeStatements) {
-      for (org.opensaml.saml.saml2.core.Attribute attribute : curStatement.getAttributes()) {
-        Collection<String> attributeNames =
-            SYS_HIGH_TO_METACARD_ATTRIBUTE_MAPPING.get(attribute.getName());
-
-        if (attributeNames == null || attributeNames.isEmpty()) {
-          continue;
+  @NotNull
+  private static Map<String, List<String>> getHighWaterSecurityMarkings(
+      Map<String, String> properties) throws DefaultSecurityAttributeValuesPluginException {
+    final String karafEtcSystemPropertyName = "karaf.etc";
+    final String karafEtcSystemPropertyValue = System.getProperty(karafEtcSystemPropertyName);
+    if (karafEtcSystemPropertyValue == null) {
+      throw new DefaultSecurityAttributeValuesPluginException(
+          "Unable to get " + karafEtcSystemPropertyName + " system property value");
+    }
+
+    final String userAttributesFilePath =
+        karafEtcSystemPropertyValue + File.separator + USERS_ATTRIBUTES_FILE_NAME;
+    final JsonElement usersDotAttributesFileJsonElement;
+    try (final Reader reader = new FileReader(userAttributesFilePath)) {
```

Comment:
✏️ This might look a lot cleaner if the concern of parsing the users.attributes file was broken out into a separate class.

Reply:
Good suggestion. Maybe when this code it touched again, we can consider refactoring this out.